providers/implementations/keymgmt/mac_legacy_kmgmt.c.in

```diff
@@ -297,79 +353,76 @@ static int mac_export(void *keydata, int selection, OSSL_CALLBACK *param_cb,
     return ret;
 }
 
-static const OSSL_PARAM mac_key_types[] = {
-    OSSL_PARAM_octet_string(OSSL_PKEY_PARAM_PRIV_KEY, NULL, 0),
-    OSSL_PARAM_utf8_string(OSSL_PKEY_PARAM_PROPERTIES, NULL, 0),
-    OSSL_PARAM_END
-};
-static const OSSL_PARAM *mac_imexport_types(int selection)
-{
-    if ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0)
-        return mac_key_types;
-    return NULL;
-}
+#define mac_get_params_st mac_common_params_st
 
-static const OSSL_PARAM cmac_key_types[] = {
-    OSSL_PARAM_octet_string(OSSL_PKEY_PARAM_PRIV_KEY, NULL, 0),
-    OSSL_PARAM_utf8_string(OSSL_PKEY_PARAM_CIPHER, NULL, 0),
-    OSSL_PARAM_utf8_string(OSSL_PKEY_PARAM_ENGINE, NULL, 0),
-    OSSL_PARAM_utf8_string(OSSL_PKEY_PARAM_PROPERTIES, NULL, 0),
-    OSSL_PARAM_END
-};
-static const OSSL_PARAM *cmac_imexport_types(int selection)
+{- produce_param_decoder('mac_get_params',
+                         (['PKEY_PARAM_PRIV_KEY', 'key', 'octet_string'],
+                         )); -}
+
+static int mac_get_params(void *keydata, OSSL_PARAM params[])
 {
-    if ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0)
-        return cmac_key_types;
-    return NULL;
+    struct mac_common_params_st p;
+    MAC_KEY *key = keydata;
+
+    if (key == NULL || !mac_get_params_decoder(params, &p))
+        return 0;
+
+    return key_to_params(key, NULL, &p);
 }
 
-static int mac_get_params(void *key, OSSL_PARAM params[])
+static const OSSL_PARAM *mac_gettable_params(void *provctx)
 {
-    return key_to_params(key, NULL, params);
+    return mac_get_params_list;
 }
 
-static const OSSL_PARAM *mac_gettable_params(void *provctx)
+#define cmac_get_params_st mac_common_params_st
+
+{- produce_param_decoder('cmac_get_params',
+                         (['PKEY_PARAM_PRIV_KEY', 'key', 'octet_string'],
+                          ['PKEY_PARAM_CIPHER',     'cipher', 'utf8_string'],
```

Comment:
nit: indentation looks strange here compared to previous line. Same with line below.

Reply:
Fixed.